Essentials/src/main/java/com/earth2me/essentials/utils/AdventureUtil.java

```diff
@@ -52,4 +84,36 @@ public static String minifyLegacy(String text) {
 
         return buffer.toString();
     }
+
+    public static NamedTextColor fromChar(final char c) {
+        final int index = LOOKUP.indexOf(c);
+        if (index == -1 || index > 15) {
+            return null;
+        }
+        return COLORS[index];
+    }
+
+    public static Placeholder parsed(final String literal) {
+        return new Placeholder(literal);
+    }
+
+    public static Tag supplyTag(final boolean primary) {
+        if (primary) {
+            return ess != null ? ess.getSettings().getPrimaryColor() : Tag.styling(NamedTextColor.GOLD);
+        }
+        return ess != null ? ess.getSettings().getSecondaryColor() : Tag.styling(NamedTextColor.RED);
+    }
+
+    public static class Placeholder {
```

Comment:
This could do with a more descriptive name:
- `ParsedPlaceholder`?
- `MiniMessagePlaceholder`? (feels way too ambiguous)